bread/layout/components/datatable.py

```diff
@@ -227,6 +226,19 @@ def with_toolbar(
                 ),
             )
 
+        searchbar = Search(
+            widgetattributes={
+                "autofocus": True,
+                "name": search_urlparameter,
+                "value": hg.F(
+                    lambda c: html.escape(c["request"].GET.get(search_urlparameter, ""))
+                ),
+                "onfocus": "this.setSelectionRange(this.value.length, this.value.length);",
+            }
+        )
+        searchbar[0][3].attributes[
```

Comment:
I think it would be appropriate to have a field `searchbar.close_button` instead of these magic indices

Reply:
I would actually prefer adding close_button_attributes to the constructor parameters of Search (I admit to be a fan of immutability).

Reply:
This is even harder to understand for new developers to continue.

Reply:
Okay, so I refactored this quite a bit, this should look better.

Regarding the close button: I prefere  a ```close_button``` attribute over an additional parameter ```close_button_attributes```, as it is an easier to use API. In general I would not try to enforce immutability on components, mainy because htmlgenerator and Python are both not designed for immutability. It allows for an easier implementation and API in many cases.

Reply:
I think every small step towards immutability helps. But I think we can just agree that we disagree ;)

Reply:
:D Okay, let's leave it at that.